autoload/phpactor.vim

```diff
@@ -314,23 +314,22 @@ endfunction
 """""""""""""""""""""""
 " Utility functions
 """""""""""""""""""""""
+function! s:isOpenInCurrentWindow(filePath)
+  return expand('%:p') == a:filePath
+endfunction
 
 function! phpactor#_switchToBufferOrEdit(filePath)
-    if expand('%:p') == a:filePath
-        " filePath is currently open
+    if <SID>isOpenInCurrentWindow(a:filePath)
```

Comment:
what is <SID> ?

Reply:
When you use a function that is defined as local to the script (`function s:foo()`), then using <SID> at the call as the prefix allows to avoid ambiguities especially when there exist a local function with the same name and both are run (non directly) from any external (as usually when using autoloaded functions) script. See `:help SID`.

From the technical point of view local functions are not fully protected from running from outside the script. If you run `:function` to see the list of defined functions then you may notice that local functions are also listed there. For example, at the moment I see a line `<SNR>120_string_encode(str) abort` (probably from vim-unimpaired)
and I can use it by `echo <SNR>120_string_encode('zxc')`.

Reply:
So if I understand correctly, we use the `s:` scope to make this a "private" function. I don't understand why we can't call `s:isOpenCurrentWindow` instead of `<SID>isOpenCurrentWindow` though?

Reply:
Yes, we may consider them "private" function (local to the script) remembering that we can still call it prefixed with a script number as above. 

I made basic test both on vim and neovim that suggested that using <SID> is unnecessary when a function is not called from a mapping. Indeed, the help text stands only about mapping, but it could sound more clear like [there](https://stackoverflow.com/a/16768478).

Thanks for the question.